File: app-front/src/services/study.ts
import { api } from "./api";
import { Study } from "@/types";

export interface CreateStudyPayload {
  title: string;
  verse: string;
  introduction: string;
  explanation: string;
  questions: string[];
  applicationIndividual: string;
  applicationGroup: string;
  prayerTopics: string[];
  conclusion: string;
}

export const studyService = {
  async getAll(): Promise<Study[]> {
    const { data } = await api.get<Study[]>("/study");
    return data;
  },

  async getById(id: number): Promise<Study> {
    const { data } = await api.get<Study>(`/study/${id}`);
    return data;
  },

  async create(payload: CreateStudyPayload): Promise<Study> {
    const { data } = await api.post<Study>("/study", payload);
    return data;
  },

  async delete(id: number): Promise<void> {
    await api.delete(`/study/${id}`);
  },
};

File: app-back/Models/Study.cs
namespace CelulaApp.Models;

public class Study
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Verse { get; set; } = string.Empty;
    public string Introduction { get; set; } = string.Empty;
    public string Explanation { get; set; } = string.Empty;
    public List<string> Questions { get; set; } = new();
    public string ApplicationIndividual { get; set; } = string.Empty;
    public string ApplicationGroup { get; set; } = string.Empty;
    public List<string> PrayerTopics { get; set; } = new();
    public string Conclusion { get; set; } = string.Empty;

    public ICollection<Meeting> Meetings { get; set; } = new List<Meeting>();
}
